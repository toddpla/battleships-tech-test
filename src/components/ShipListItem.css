.ship:hover {
  font-weight: bold;
  cursor: pointer;
}

.ship-selected {
  font-weight: bold;
}
