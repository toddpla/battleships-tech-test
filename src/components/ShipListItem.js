import React, { Component } from 'react';

export class ShipListItem extends Component {

  render() {
    return (
      <div>
        <h3>{this.props.name}</h3>
      </div>
    );
  }
}

export default ShipListItem;
