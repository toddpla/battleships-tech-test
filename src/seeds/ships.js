export default [{
  id: 1,
  name: 'Carrier',
  size: 5
}, {
  id: 2,
  name: 'Battleship',
  size: 4
}, {
  id: 3,
  name: 'Cruiser',
  size: 3
}, {
  id: 4,
  name: 'Submarine',
  size: 3
}, {
  id: 5,
  name: 'Destroyer',
  size: 2
}]