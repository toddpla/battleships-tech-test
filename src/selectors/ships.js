export const unSelectedShips = ships => ships.filter(ship => ship.squares.length === 0)
