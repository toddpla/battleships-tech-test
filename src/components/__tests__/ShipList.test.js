import React from 'react'
import { ShipListItem } from '../ShipListItem'
import renderer from 'react-test-renderer';
import ships from '../../seeds/ships'

test('renders correctly', () => {
  const shipListItem = renderer.create(<ShipListItem {...ships[0]}/>).toJSON();
  expect(shipListItem).toMatchSnapshot();
});
