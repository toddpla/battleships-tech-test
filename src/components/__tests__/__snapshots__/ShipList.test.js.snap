// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Carrier
  </h3>
</div>
`;
